doc/api/stream.md

```diff
@@ -1889,6 +1889,46 @@ await dnsResults.forEach((result) => {
 console.log('done'); // Stream has finished
 ```
 
+### `readable.toArray([options])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `options` {Object}
+  * `signal` {AbortSignal} allows cancelling the toArray operation if the
+    signal is aborted.
+* Returns: {Promise} a promise containing an array (if the stream is in
+  object mode) or Buffer with the contents of the stream.
```

Comment:
I'd prefer if this used a `Uint8Array` instead of `Buffer`.

Reply:
@jasnell `.read()` returns a buffer so I assumed this should return the same type. As buffers are all `Uint8Array`s and this behavior is stream specific I'm not sure what we gain?

Why do you prefer `Uint8Array` here? Is there some long-term plan to minimize/deprecate `Buffer` as a type?

Reply:
> Why do you prefer Uint8Array here? Is there some long-term plan to minimize/deprecate Buffer as a type?

Consider it a non-blocking personal preference to start moving away, as much as possible, from Node.js specific APIs where there is a language and web platform standard API option available. We can't change the existing `stream.Readable` API but we don't have to introduce new dependencies on the implementation specific APIs.

Reply:
@ronag @mcollina thoughts?

Reply:
> Consider it a non-blocking personal preference to start moving away, as much as possible, from Node.js specific APIs where there is a language and web platform standard API option available.

I strongly agree with this too, as a heavy user of Node.js. `Buffer` makes it harder to create cross-platform packages.

I think this merits a larger discussion in a dedicated issue though. I'd like to see an official stance on this, for `Buffer` specifically, but also generally.

Reply:
@sindresorhus  @jasnell opened https://github.com/nodejs/node/issues/41588 - since this API is experimental I am content with shipping it with buffer and breaking it in a small way in the next version.

Reply:
What does the spec say/do here?

Reply:
The rule of least surprise would tell me to return a Buffer here. I don't think we should return a UInt8Array unless it's specified by the spec.

Reply:
@ronag If we were to `toArray` an async iterator it would return a list of all the chunks I would assume it would not concat them. I can change the behavior to an array of buffers but I am not sure it's particularly useful to users.

I suspect/hope that web streams would overload `.toArray` to do what users would expect (return the stream as an array) rather than a list of chunks.

Reply:
This behavior is probably changing and we are going with no non-objectMode specific behavior https://github.com/nodejs/node/pull/41615 - it's also good to Note `toArray` will always return well... an array :)